src/DataMapper/Product/Provider/AttributesDataMapperValuesProvider.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\DataMapper\Product\Provider;
+
+use Setono\SyliusMeilisearchPlugin\Model\IndexableInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+final class AttributesDataMapperValuesProvider implements DataMapperValuesProviderInterface
+{
+    public function provide(IndexableInterface $source, array $context = []): array
+    {
+        Assert::keyExists($context, 'locale_code');
+        Assert::string($context['locale_code']);
+        Assert::isInstanceOf($source, ProductInterface::class);
```

Comment:
Why did you choose not to include the `IndexScope` in the signature? This feels redundant when you have the needed data on an object where this is called. What do you think?